internal/resource/execute.go

```diff
@@ -0,0 +1,117 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	operatorClientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	subClientsetv1 "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1opts "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"os"
+)
+
+type Cluster struct {
+	Config     *rest.Config
+	Name       string
+	KubeClient kubernetes.Interface
+	DynClient  dynamic.Interface
+	SubmClient subClientsetv1.Interface
+	Submariner *v1alpha1.Submariner
+}
+
+func ExecuteMultiCluster(restConfigProducer restconfig.Producer, run func(*Cluster) bool) {
+	success := true
+
+	for _, config := range restConfigProducer.MustGetForClusters() {
+		fmt.Printf("Cluster %q\n", config.ClusterName)
+
+		cluster, errMsg := NewCluster(config.Config, config.ClusterName)
+		if cluster == nil {
+			success = false
+
+			fmt.Println(errMsg)
+			fmt.Println()
+
+			continue
+		}
+
+		success = run(cluster) && success
+
+		fmt.Println()
+	}
+
+	if !success {
+		os.Exit(1)
+	}
+}
+
+func NewCluster(config *rest.Config, clusterName string) (*Cluster, string) {
+	cluster := &Cluster{
+		Config: config,
+		Name:   clusterName,
+	}
+
+	clientProducer, err := client.NewProducerFromRestConfig(config)
+
+	cluster.KubeClient = clientProducer.ForKubernetes()
+	cluster.DynClient = clientProducer.ForDynamic()
+	cluster.SubmClient = clientProducer.ForSubmariner()
+	cluster.Submariner, err = getSubmarinerResourceWithError(clientProducer.ForOperator())
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, fmt.Sprintf("Error retrieving Submariner resource: %v", err)
+	}
+
+	return cluster, ""
+}
+
+func getSubmarinerResourceWithError(operatorClient operatorClientset.Interface) (*v1alpha1.Submariner, error) {
+	// TODO skitt namespace constant
+	submariner, err := operatorClient.SubmarinerV1alpha1().Submariners("submariner-operator").
```

Comment:
As the comment suggests.

```suggestion
	submariner, err := operatorClient.SubmarinerV1alpha1().Submariners(constants.OperatorNamespace).
```

Reply:
I thought I did this.